pkgs/top-level/all-packages.nix

```
@@ -19293,6 +19293,10 @@ with pkgs;
 
   indent = callPackage ../development/tools/misc/indent { };
 
+  inochi-creator = callPackage ../applications/misc/inochi2d/creator.nix { };
+
+  inochi-session = callPackage ../applications/misc/inochi2d/session.nix { };
+
```

```suggestion
  inherit (callPackages ../applications/misc/inochi2d { })
    inochi-creator inochi-session;

```

And create `inochi2d/default.nix` to export these two attributes. That will be allowed

---

In my years of Nixpkgs, I somehow haven't come across that function/pattern. Well, I've seen the function but never bothered to check what it does.

How does this play into by-name?

---

It's not that common, but it's one of the easiest ways to group together some packages while getting `.override` for each of them (with the arguments of the `callPackages` file).

Admittedly this failure case wasn't fully thought out in RFC 140, but I think it works out fine, because it requires having some way to distinguish fully independent packages from ones that share some internal functionality.